tests/utils/test_client.py

```diff
@@ -53,4 +58,33 @@ def test_client_enforce_healthy():
     assert my_client.port == int(CONFIG["my_instance_name"]["port"])
     assert my_client.protocol == CONFIG["my_instance_name"]["protocol"]
     assert my_client.base_path == "/api/versioned/v1/"
-    assert my_client.auth == UsernamePasswordAuth("admin", os.environ["TAMR_TOOLBOX_PASSWORD"],)
+    assert my_client.auth == UsernamePasswordAuth(
+        "admin", os.environ["TAMR_TOOLBOX_PASSWORD"],
+    )
+
+
+@mock_api()
+def test_get_with_connection_retry():
+    # To ensure timeout error is raised to test correct ConnectionError is caught
+
+    log_prefix = "caught_connection_error"
+    log_file_path = os.path.join(
+        tempfile.gettempdir(), f"{log_prefix}_{utils.logger._get_log_filename()}"
+    )
```

Comment:
I'm actually learning that this method of getting a tempdir is not correct.  Instead we should be doing
```suggestion
    with tempfile.TemporaryDirectory() as tempdir:
        log_file_path = os.path.join(
            tempdir.name, f"{log_prefix}_{utils.logger._get_log_filename()}"
        )
```
The rest of the test also needs to be indented under this `with` context manager.